lib/net/vnc.rb

```diff
@@ -133,9 +133,12 @@ def connect
 			socket.write((options[:shared] ? 1 : 0).chr)
 
 			# ServerInitialisation
+			framebuffer_width  = socket.read(2).to_s.unpack('n')[0].to_i
```

Comment:
Not entirely clear on what the to_s and to_i are for - is the idea that if the socket is eof then framebuffer_width will be 0 (nil => '' unpacked to nil => 0?). Could perhaps just do `width, height = socket.read(4).unpack('n2')`, but maybe when your helper things land this gets replaced with those anyway...

Reply:
Ouch, that is a good point. I will see what I can do.